Tools/paletteable sprite editor/js/marvinj-0.9.js

```diff
@@ -0,0 +1,2665 @@
+
+function MarvinColorModelConverter(){}
+
+MarvinColorModelConverter.rgbToBinary = function(img, threshold){		
+	var resultImage = new MarvinImage(img.getWidth(), img.getHeight(), MarvinImage.COLOR_MODEL_BINARY);
+
+	for(var y=0; y<img.getHeight(); y++){
+		for(var x=0; x<img.getWidth(); x++){
+			var gray = Math.ceil(((img.getIntComponent0(x, y)*0.3)+(img.getIntComponent1(x, y)*0.59)+(img.getIntComponent2(x, y)*0.11)));
+			
+			if(gray <= threshold){
+				resultImage.setBinaryColor(x, y, true);
+			}
+			else{
+				resultImage.setBinaryColor(x, y, false);
+			}
+		}
+	}
+	return resultImage;
+};
+
+MarvinColorModelConverter.binaryToRgb = function(img){
+	var resultImage = new MarvinImage(img.getWidth(), img.getHeight(), MarvinImage.COLOR_MODEL_RGB);
+
+	for(var y=0; y<img.getHeight(); y++){
+		for(var x=0; x<img.getWidth(); x++){
+			if(img.getBinaryColor(x, y)){
+				resultImage.setIntColor(x, y, 255, 0,0,0);
+			}
+			else{
+				resultImage.setIntColor(x, y, 255, 255,255,255);
+			}
+		}
+	}
+	return resultImage;
+};
+	
+MarvinColorModelConverter.rgbToHsv = function(rgbArray){
+	var hsvArray = new Array(rgbArray.length*3);
+	
+	var red,green,blue;
+	for(var i=0; i<rgbArray.length; i++){
+		red = (rgbArray[i] & 0xFF0000) >>> 16;
```

Comment:
Codacy found an issue: [The numeric literal '0xFF0000' will have at different value at runtime.](https://app.codacy.com/gh/unitystation/unitystation/pullRequest?prid=6395467)